### PR TITLE
Run tests on push to version branches + pre-commit settings

### PR DESCRIPTION
**Pull request by DaveSys911**
* Added regex to run on push ci to include version branches
* Added pre-commit conf file
* editorconfig file
* mage config hook

### CODE REVIEW

**Review by oren-zohar — CHANGES_REQUESTED** [review]
can you please add an explanation to the readme about how to set up the pre-commit hooks?
does it run the hooks on the diff or the entire repo? wandering how the flake8 and pep8 passed for you

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  - repo: https://github.com/pycqa/pylint
+    rev: pylint-2.6.0
+    hooks:
+      - id: pylint
+        args: [
+            "-rn", # Only display messages
+            "-sn", # Don't display the score
+            "--rcfile=tests/pylintrc", # Link to your config file
+          ]
```
Comment (gurevichdmitry) [inline .pre-commit-config.yaml:46]: @DaveSys911 very very good.

**Review by gurevichdmitry — APPROVED** [review]
@DaveSys911  LGTM for .pre-commit-config.yaml -> Python section

**Comment by DaveSys911** [comment]
> can you please add an explanation to the readme about how to set up the pre-commit hooks? does it run the hooks on the diff or the entire repo? wandering how the flake8 and pep8 passed for you

Only on diffs by default, u can run it on all/some files on demand like this `pre-commit run <hook_name>  --all-files --verbose`

**Comment by oren-zohar** [comment]
> Only on diffs by default, u can run it on all/some files on demand like this `pre-commit run <hook_name> --all-files --verbose`

great, can you please add it to the readme as we have [here](https://github.com/elastic/csp-security-policies#pre-commit-hooks)?

**Comment by DaveSys911** [comment]
> > Only on diffs by default, u can run it on all/some files on demand like this `pre-commit run <hook_name> --all-files --verbose`
> 
> great, can you please add it to the readme as we have [here](https://github.com/elastic/csp-security-policies#pre-commit-hooks)?

Done. Also went ahead & added .editorconfig file & corresponding readme entry and a mage config hook. It will fail commits with manual changes to cloudbeat.yml (without using .tmpl files & running mage config).

File: README.md [hunk 2]
```diff
@@ -23,7 +23,7 @@
 ```zsh
 just setup-env
 ```
-
+TEST
```
Comment (oren-zohar) [inline README.md:26]: please remove

**Review by oren-zohar — CHANGES_REQUESTED** [review]

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**